e2e/cgroups/cgroups.go

```diff
@@ -41,6 +41,78 @@ type ctx struct {
 	env e2e.TestEnv
 }
 
+// instanceStats tests an instance ability to output stats
+func (c *ctx) instanceStats(t *testing.T, profile e2e.Profile) {
+	e2e.EnsureImage(t, c.env)
+
+	// All tests require root
+	tests := []struct {
+		name           string
+		createArgs     []string
+		startErrorCode int
+		statsErrorCode int
+	}{
+		{
+			name:           "basic stats create",
+			createArgs:     []string{"--apply-cgroups", "testdata/cgroups/cpu_success.toml", c.env.ImagePath},
+			statsErrorCode: 0,
+			startErrorCode: 0,
+		},
+	}
+
+	for _, tt := range tests {
+
+		// stats only for privileged atm
+		if !profile.Privileged() {
+			t.Skip()
+		}
+
+		// We always expect stats output, not create
+		createExitFunc := []e2e.SingularityCmdResultOp{}
+		instanceName := randomName(t)
+
+		// Start the instance with cgroups for stats
+		createArgs := append(tt.createArgs, instanceName)
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name+"/start"),
```

Comment:
Instead of putting `/` delimited subtests here, it's more common to wrap everything inside the loop in an `t.Run(` call with an anonymous function, so that the structure of the nested tests is more apparent.

i.e.

```
for _, tt := range tests {
    t.Run(tt.name, func(t *.testing.T) {
          ...
          c.env.RunSingularity(
              t,
              e2e.AsSubtest("start"),
              ....
    })
}